homeassistant/components/system_bridge/config_flow.py

```diff
@@ -56,15 +57,13 @@ async def validate_input(
     try:
         async with async_timeout.timeout(30):
             await websocket_client.connect(session=async_get_clientsession(hass))
-            await websocket_client.get_data(["system"])
-            while True:
-                message = await websocket_client.receive_message()
-                _LOGGER.debug("Message: %s", message)
-                if (
-                    message[EVENT_TYPE] == TYPE_DATA_UPDATE
-                    and message[EVENT_MODULE] == "system"
-                ):
-                    break
+            hass.async_create_task(websocket_client.listen())
+            response = await websocket_client.get_data(GetData(modules=["system"]))
+            _LOGGER.info("Got response: %s", response.json())
+            try:
+                system = System(**response.data)
```

Comment:
The library user shouldn't need to select what models to apply to a result. This should be handled by the library.

Reply:
Done